server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -1434,7 +1434,25 @@ private void writeConditionalMutations(Map<KeyExtent,List<ServerConditionalMutat
         long t2 = System.currentTimeMillis();
         updateAvgCommitTime(t2 - t1, sendables.size());
       }
+    }
+
+    /**
+     * Return true if and only if the attempt has at least one mutation that violated constraints,
+     * and either no valid mutations, or some valid mutations and the commit session was available
+     * from the tablet's memory.
+     */
+    private boolean
+        hasViolationsAndMayHaveWritableNonViolators(final TabletMutationPrepAttempt attempt) {
+      return attempt.hasViolations() && (!attempt.hasNonViolators() || attempt.hasCommitSession());
+    }
+
+    private void addMutationsAsTCMResults(final List<TCMResult> list,
+        final Collection<? extends Mutation> mutations, final TCMStatus status) {
+      mutations.stream().map(mutation -> toTCMResult(mutation, status)).forEach(list::add);
+    }
 
+    private TCMResult toTCMResult(final Mutation mutation, final TCMStatus status) {
+      return new TCMResult(((ServerConditionalMutation) mutation).getID(), status);
```

Comment:
Could inline this, since it's only used once. If doing so would make the lambda less readable in `addMutationsAsTCMResults`, you could use the curly brace form so the lambda used in the map function wraps to the next line better.

Reply:
It's not a problem to inline it.